modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/TopLevelParameterTypeFactoryBean.java

```diff
@@ -27,16 +28,18 @@
 {
 
     private final ValueResolver valueResolver;
+    private final MuleContext muleContext;
 
-    TopLevelParameterTypeFactoryBean(ElementDescriptor element, DataType dataType)
+    TopLevelParameterTypeFactoryBean(ElementDescriptor element, DataType dataType, MuleContext muleContext)
     {
         this.valueResolver = XmlExtensionParserUtils.parseElement(element, EMPTY, dataType, null);
+        this.muleContext = muleContext;
     }
 
     @Override
     public Object getObject() throws Exception
     {
-        return valueResolver.resolve(VoidMuleEvent.getInstance());
+        return valueResolver.resolve(getInitialiserEvent(muleContext));
```

Comment:
actually I can't...but it was about that you don't know if the ValueResolver will need to use the event's muleContext and VoidMuleEvent would fail in such a scenario.... Now, I need this initializer event anyways in the ConfigurationBuilder so it's good to change it to fix that original comment plus have consistency